Kuky/app/src/main/java/com/example/keenan/kuky/adapters/KuCardAdapter.java

```diff
@@ -0,0 +1,191 @@
+            public void onClick(View v) {
+                int userId = getUserId();
+                int kuId = mKu.getId();
+                Log.d(TAG, new KuActionRequest(userId, kuId).toString());
+                if (userId > 0) {
+                    ApiClient.getKukyApiClient().downvoteKu(new KuActionRequest(userId, kuId))
+                            .subscribeOn(Schedulers.newThread())
+                            .observeOn(AndroidSchedulers.mainThread())
+                            .subscribe(new Subscriber<KuActionResponse>() {
+                                @Override
+                                public void onCompleted() {
+
+                                }
+
+                                @Override
+                                public void onError(Throwable e) {
+
+                                }
+
+                                @Override
+                                public void onNext(KuActionResponse kuActionResponse) {
+                                    Log.d(TAG, kuActionResponse.getStatus());
+                                }
+                            });
+                }
+            }
+        });
+
+        holder.vFavorite.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                int userId = getUserId();
+                int kuId = mKu.getId();
+                Log.d(TAG, new KuActionRequest(userId, kuId).toString());
+                if (userId > 0) {
+                    ApiClient.getKukyApiClient().favoriteKu(new KuActionRequest(userId, kuId))
+                            .subscribeOn(Schedulers.newThread())
+                            .observeOn(AndroidSchedulers.mainThread())
+                            .subscribe(new Subscriber<KuActionResponse>() {
+                                @Override
+                                public void onCompleted() {
+
+                                }
+
+                                @Override
+                                public void onError(Throwable e) {
+
+                                }
+
+                                @Override
+                                public void onNext(KuActionResponse kuActionResponse) {
+                                    Log.d(TAG, kuActionResponse.getStatus());
+                                }
+                            });
+                }
+            }
+        });
+    }
+
+    public int getUserId() {
+        SharedPreferences settings = mContext.getSharedPreferences(LoginActivity.PREFS_NAME, 0);
+        return settings.getInt("userId", -1);
+    }
+
+    // Return the size of your dataset (invoked by the layout manager)
+    @Override
+    public int getItemCount() {
+        return mDataset.size();
+    }
+
+    @Override
+    public void onAttachedToRecyclerView(RecyclerView recyclerView) {
+        super.onAttachedToRecyclerView(recyclerView);
+    }
+
+    public void setList(ArrayList<Ku> newList) {
```

Comment:
Looks like we both implemented this - I read somewhere that notifydatasetchanged is better called from the fragment, but that could be wrong